src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -329,7 +329,19 @@ private static string GetVsRootFromMSBuildAssembly(string msBuildAssembly)
 
         private static string GetMSBuildExeFromVsRoot(string visualStudioRoot)
         {
-            return FileUtilities.CombinePaths(visualStudioRoot, "MSBuild", CurrentToolsVersion, "Bin", "MSBuild.exe");
+            return FileUtilities.CombinePaths(
+                visualStudioRoot,
+                "MSBuild",
+                CurrentToolsVersion,
+                "Bin",
+#if NET35
+                IntPtr.Size == 8
+#else
+                System.Runtime.InteropServices.Marshal.SizeOf<IntPtr>() == 8
+#endif
```

Comment:
`IntPtr.Size == 8` would work for new versions or older versions, right? Could we just use that version instead?

Reply:
that's fair and cleaner-looking